source/choosers.c

```diff
@@ -4,122 +4,175 @@
 #include "SystemInteractors.h"
 #include <math.h>
 
-BOOL CheckFoldersAnswer(LPWSTR pPath, LPWIN32_FIND_DATA_ARRAYW pFiles_arr_t, LPWSTR pAnswer) {
+BOOL UserIODrives(
+	LPWSTR pPath
+)
+{
+	wprintf(L"Please Choose a Drive\n");
+	WCHAR pAnswer[2];
+	wscanf_s(L"%1s", pAnswer, 2);
+	pAnswer[0] = towupper(pAnswer[0]);
+	BOOL result = CACDrives(pPath, &pAnswer);
+	return result;
+
+}
+
+BOOL UserIOFolders(
+	LPWSTR pPath,
+	LPWIN32_FIND_DATA_ARRAYW pFiles_arr_t
+)
+{
+	UserAnswer_t Answer_t = { NULL };
+	LPWSTR pAnswer[MAX_PATH] = { L'\0' };
+	wscanf_s(L"%259s", pAnswer, MAX_PATH);
+	Answer_t.data = &pAnswer;
+	Answer_t.length = wcslen(pAnswer);
+	CheckFoldersAnswer(
+		pPath,
+		pFiles_arr_t,
+		&Answer_t
+	);
+	return UserIOTraverseFolders();
+}
+
+BOOL CheckFoldersAnswer(
+	LPWSTR pPath, 
+	LPWIN32_FIND_DATA_ARRAYW pFiles_arr_t, 
+	pUserAnswer_t pAnswer_t
+) 
+{
 	//Calculating How Manny Possible characters can be numbers that represent an index and the highest value of each indavidual char.
-	int remainder = pFiles_arr_t->count;
-	int OrderOfMagnitduted = floor(log10(pFiles_arr_t->count));
-	int Power2Raise2 = OrderOfMagnitduted;
-	int* file_index = (int*)malloc(((OrderOfMagnitduted + 1) * sizeof(int)) + 1);
+	int remainder = pFiles_arr_t->count;//Checking The count Data Member Of The Files Array Struct
```

Comment:
the correct and recommended style for comments (throughout any project) is with a space: before the `//` and a space after:
```suggestion
	int remainder = pFiles_arr_t->count; // Checking The count Data Member Of The Files Array Struct
```

makes it more readable